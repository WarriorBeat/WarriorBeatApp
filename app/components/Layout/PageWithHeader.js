/**
 * PageWithHeader.js
 * PageWIthHeader Layout
 * components/Layout
 */

import React from "react"
import { PropTypes } from "prop-types"
import {
  Header, Content, Container, Left, Body, Title,
} from "native-base"
import { Icon } from "react-native-elements"
import { icons } from "config/styles"
import { Typo } from "components/Text"
import { observer, inject, PropTypes as MobxTypes } from "mobx-react/native"
import { Colors } from "components/styles"
import { StyleSheet, ViewPropTypes } from "react-native"

const styles = ({ backgroundColor }) => StyleSheet.create({
  default: {
    ...Colors.bgColor(backgroundColor),
  },
})

@inject("uiStore")
@observer
class PageWithHeader extends React.Component {
  render() {
    const {
      uiStore, children, title, containerStyle,
    } = this.props
    return (
      <Container style={[styles(this.props).default, containerStyle]}>
        <Header transparent>
          <Left>
            <Icon
              {...icons.arrow_back}
              color={Colors.type.black}
              onPress={() => uiStore.goBack()}
            />
          </Left>
          <Body>
            <Title>
              <Typo.Header>{title}</Typo.Header>
            </Title>
          </Body>
        </Header>
        <Content>{children}</Content>
      </Container>
    )
  }
}

PageWithHeader.wrappedComponent.propTypes = {
  uiStore: MobxTypes.observableObject.isRequired,
}

PageWithHeader.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
  backgroundColor: PropTypes.string,
  containerStyle: ViewPropTypes.style,
}

PageWithHeader.defaultProps = {
  title: "Header",
  backgroundColor: Colors.type.soft.white,
  containerStyle: {},
}

export default PageWithHeader
