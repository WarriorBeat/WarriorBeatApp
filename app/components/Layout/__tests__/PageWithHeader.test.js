/**
 * __tests__/PageWithHeader.test.js
 * Component Test
 * tests
 */
/* eslint-disable */

import React from "react"
import { shallow } from "react-native-testing-library"
import PageWithHeader from "../PageWithHeader"
import Providers from "tests"

test("should render correctly", () => {
  const tree = shallow(
    <Providers>
      <PageWithHeader title="Header" />
    </Providers>,
  )
  expect(tree).toMatchSnapshot()
})
