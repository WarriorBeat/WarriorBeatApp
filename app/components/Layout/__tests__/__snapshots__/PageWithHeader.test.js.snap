// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": undefined,
      },
      Object {},
    ]
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "transparent",
          "borderBottomWidth": 0.5,
          "elevation": 0,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": 0,
        }
      }
      transparent={true}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 24,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              Ôèè
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": undefined,
                "fontStyle": "normal",
                "fontWeight": "700",
                "includeFontPadding": true,
                "textTransform": undefined,
              }
            }
          >
            Header
          </Text>
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
